### PR TITLE
Add minitest support

### PR DESCRIPTION
**Pull request by markedmondson**
From #30, this small change should allow for Minitest support using the same readme instructions as for RSpec.

The only differences are scope is unsupported (Minitest doesn't really have this concept) so the object is always created immediately and destroyed after the test run. I copied a couple of the basic tests into the suite to demonstrate the functionality.

Open to critical feedback, I don't love the new private method name and someone with better knowledge of Minitest internals might be able to get it to perform more similarly to RSpec (there is the suggestion of similar execution blocks [here](https://www.rubydoc.info/github/seattlerb/minitest/Minitest/Test/LifecycleHooks#before_setup-instance_method))

### CODE REVIEW

File: lib/with_model.rb [hunk 1]
```diff
@@ -7,36 +7,46 @@
...
+  def setup_object(object, scope = nil)
+    if Module.const_defined? 'RSpec'
+      before(*scope) do
+        object.create
+      end
+
+      after(*scope) do
+        object.destroy
+      end
+    elsif Module.const_defined? 'Minitest'
```
Comment (jpgeek) [inline lib/with_model.rb:44]: Won't this also catch Minitest::Spec?

Comment (markedmondson) [reply lib/with_model.rb:44]: Yeah, good point - very likely, just pure luck I put the RSpec one first. Can you think of a better way to determine which test suite is executing?

Comment (jpgeek) [reply lib/with_model.rb:44]: I can't.  May be able to check for `Minitest::Spec` first, then fall back to `Minitest` but even if that happens to work now it seems like this is getting too far into the internals of Minitest.

Could add a `test_suite` parameter to `with_model` and default to `:rspec`.  

Could also encourage overwriting `setup_object`.  That way the developer can use whatever callbacks with whatever parameters, or even something like:
```
def setup_object(object, **args)
  object.create
  yield
  object.destroy
end
```

Maybe the intention is to use the DSL API instead in this case though?

**Comment by markedmondson** [comment]
Happy Birthday to this PR! 🍰 

I used a parameter to toggle the test runner in cb878cc9b8afb2fb969347af11c3d0de09b13db7

It also includes a minitest test suite (`rake test`) and ensures that with_model works with minitest when there are two models defined (previously one would overwrite the other).

**Comment by miks** [comment]
Something is wrong with Travis CI jobs although master ir green. Besides that, I think it's time to merge this finally :)

**Comment by markedmondson** [comment]
Ah, I'd left a require in the test for pry which I was using for debugging. Just pushed a change which hopefully pleases Travis.

**Comment by markedmondson** [comment]
Hmm just failing on jruby now, not something I'm overly familiar with - anyone able to pitch in?

**Comment by miks** [comment]
@markedmondson can you add something like this https://github.com/carrierwaveuploader/carrierwave/commit/6e6f3472a2243819790fe994f756c88a7e0764ad#diff-89eebfcbc0f14b6d989517837ca1e94fce4e2ce9a03233641cd936f2b8d2ed94R4 to fix jruby error?

**Comment by markedmondson** [comment]
> @markedmondson can you add something like this [carrierwaveuploader/carrierwave@6e6f347#diff-89eebfcbc0f14b6d989517837ca1e94fce4e2ce9a03233641cd936f2b8d2ed94R4](https://github.com/carrierwaveuploader/carrierwave/commit/6e6f3472a2243819790fe994f756c88a7e0764ad#diff-89eebfcbc0f14b6d989517837ca1e94fce4e2ce9a03233641cd936f2b8d2ed94R4) to fix jruby error?

Thanks - tried every combination and still can't get it to pass for this case: https://travis-ci.org/github/Casecommons/with_model/jobs/760232785

**Comment by miks** [comment]
@markedmondson move jruby conditional require to the start of the page (before `require 'simplecov'`)

**Comment by markedmondson** [comment]
Thanks @miks, that got it - curious what was triggering the premature call to i18n? (Especially only in 5.2)

I figured it just needed to be before the activemodel require...

**Comment by miks** [comment]
@markedmondson I didn't investigate it further, only got the idea that something is doing it already

**Comment by markedmondson** [comment]
Cool.

Thanks for the hand holding and prompt support on this! :)

**Pull request merged**